determined-ai/determined
# feat: Display metrics in a grid of charts [WEB-653]

## Description

A grid of charts, with one chart for all metric, replaces the selected-metrics chart + workloads table.

Key elements:
- react-window FixedSizeList is used on the LogViewer infinite scroll; we use their FixedSizeGrid here
- react-window recommends use of AutoSizer from react-virtualized-auto-sizer to give us a px width for the component (replacement for passing width={"100%"})
- we have some flexibility to style the chart blocks and give them some margins, unfortunately FixedSizeGrid is managing the actual placement with px coordinates, so it is not as standard as CSS grid in Workspace cards.

Removing the workloads table allows removing some code from types and utils, but not API.

## Test Plan

- Open a single-trial experiment which has been successful. Garrett has provided a 200-metric experiment for testing multi-metrics
- Confirm that we have a scrollable list of all metrics
- Confirm that a dropdown above the charts still switches from Linear -> Log scale on all charts
- Confirm that the metrics are a responsive grid - in a narrow window we have 1 chart per row, at a wider window we have 2 charts (min 500px per chart), then wider -> 3 charts (on a laptop, use Cmd - to zoom-out and increase screen width)

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
